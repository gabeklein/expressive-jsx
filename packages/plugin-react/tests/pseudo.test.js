const test = require("./_adapter");

describe("pseudo", () => {
  test("before/after", `
    () => do {
      before: {
        content: "This is content"
        color: blue;
      }

      after: {
        background: red;
        color: white;
      }

      <div>Hello</div>
    }
  `);

  test("nthOfType", `
    () => do {
      nthOfType: {
        select: "2n"
        color: blue;
      }
      
      nthOfType: {
        select: "3n+2"
        color: red;
      }

      <div>Hello</div>
    }
  `);
});