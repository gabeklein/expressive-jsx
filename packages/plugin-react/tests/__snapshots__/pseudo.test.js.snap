// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pseudo before/after 1`] = `
"() => create(\\"div\\", {
  \\"className\\": \\"do_erd\\"
}, create(\\"div\\", {}, \\"Hello\\"));

Styles.include(\`
	.do_erd::before { 
		content: \\"This is content\\";
		color: blue;
	}
	.do_erd::after { 
		background: red;
		color: white;
		content: \\"\\";
	}
\`);"
`;

exports[`pseudo before/after 2`] = `
"() => <div className=\\"do_erd\\"><div>Hello</div></div>;

Styles.include(\`
	.do_erd::before { 
		content: \\"This is content\\";
		color: blue;
	}
	.do_erd::after { 
		background: red;
		color: white;
		content: \\"\\";
	}
\`);"
`;

exports[`pseudo nthOfType 1`] = `
"() => create(\\"div\\", {
  \\"className\\": \\"do_erd\\"
}, create(\\"div\\", {}, \\"Hello\\"));

Styles.include(\`
	.do_erd:nth-of-type(2n) { 
		color: blue;
	}
	.do_erd:nth-of-type(3n+2) { 
		color: red;
	}
\`);"
`;

exports[`pseudo nthOfType 2`] = `
"() => <div className=\\"do_erd\\"><div>Hello</div></div>;

Styles.include(\`
	.do_erd:nth-of-type(2n) { 
		color: blue;
	}
	.do_erd:nth-of-type(3n+2) { 
		color: red;
	}
\`);"
`;
